components/CountryNavigator.js

```diff
@@ -0,0 +1,248 @@
+import { useEffect, useState, useMemo } from "react";
+import Link from "next/link";
+import { ArrowBack } from "@mui/icons-material";
+import { scaleLinear, scaleBand, scaleOrdinal } from "@visx/scale";
+import { format } from "d3-format";
+import Head from "next/head";
+import { flag } from "country-emoji";
+import Box from "@mui/material/Box";
+import { motion } from "framer-motion";
+import IconButton from "@mui/material/IconButton";
+import Button from "@mui/material/Button";
+import { styled } from "@mui/material/styles";
+import PlayCircleOutlineIcon from "@mui/icons-material/PlayCircleOutline";
+import PauseCircleOutlineIcon from "@mui/icons-material/PauseCircleOutline";
+import ArrowCircleLeftOutlinedIcon from "@mui/icons-material/ArrowCircleLeftOutlined";
+import ArrowCircleRightOutlinedIcon from "@mui/icons-material/ArrowCircleRightOutlined";
+
+import { nth } from "../utils";
+import { RainbowLink } from "../components/RainbowLink";
+import Legend from "../components/Legend";
+import { CATEGORIES_ORDERED_LIST, RAINBOW_COLORS } from "../constants";
+import { getData } from "../data";
+import { useFlagDimensions } from "../hooks/useFlagDimensions";
+import { Flag } from "../components/Flag";
+
+const CountryNavigator = () => {
+  const [year, setYear] = useState(2024);
+  const [sortDict, setSortDict] = useState({
+    type: "ranking",
+    ascending: true,
+  });
+
+  const data = useMemo(
+    () => getData({ years: [year], sortingParams: sortDict }),
+    [year, sortDict]
+  );
+
+  const countries = data.map((d) => d.country);
+  const [currentCountryIndex, setCurrentCountryIndex] = useState(0);
+  const dataCountry = useMemo(() => getData({ countries: countries }), [
```

Comment:
```suggestion
  const dataCountry = useMemo(() => getData({ countries: [currentCountry] }), [
   currentCountry])
```
This way, we only get the data for the country we need, and because of the `useMemo`, it only runs when `currentCountry` changes. Additionally, we avoid filtering the data on every render, like was happening on line :62